Include value type when unable to dump a service container

ruudk opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      |no
| New feature?  | yes
| Deprecations? |no 
| Tickets       | 
| License       | MIT
| Doc PR        | 

When your Definition contains invalid values, like an object or resource, an error is thrown.

Often, it's hard to tell what causes it. By showing the type of the value, it can help developers
resolve the problem faster.

nicolas-grekas commented on src/Symfony/Component/DependencyInjection/CHANGELOG.md:
```diff
@@ -14,6 +14,7 @@ CHANGELOG
  * Add `#[AsAlias]` attribute to tell under which alias a service should be registered or to use the implemented interface if no parameter is given
  * Allow to trim XML service parameters value by using `trim="true"` attribute
  * Allow extending the `Autowire` attribute
+ * Include value type when unable to dump a service container
```
should be removed as we're going to merge this as "minor" (aka "not a feature nor a bug")

ruudk replied:
Done

ruudk committed:
Print value type on dump error

When your Definition contains invalid values, like an object or resource, an error is thrown.

Often, it's hard to tell what causes it. By showing the type of the value, it can help developers
resolve the problem faster.

ruudk commented:
@nicolas-grekas Shouldn't I change the target branch?

nicolas-grekas commented:
Thank you @ruudk.

nicolas-grekas commented:
6.3 is correct, this is still an improvement.
